### PR TITLE
feat(corpus_importer): Implements data import into S3 storage

### PR DESCRIPTION
**Pull request by ERosendo**
This PR introduces a new command `customer_import_project` to retrieve data from a source (prod database or replica) and creates manifest files listing those records in Amazon S3. These manifest files can then be used by batch processing jobs (e.g., AWS Glue, AWS Batch) to efficiently process the data in parallel.

Users can choose which record type to import:

- `SEARCH_TYPES.RECAP_DOCUMENT`: Imports recap document records.
- `SEARCH_TYPES.OPINION`: Imports opinion records.
- `SEARCH_TYPES.ORAL_ARGUMENT`: Imports oral argument records (not implemented yet).

The command requires the following arguments:

- `--record-type`: The type of records to import (**required**).
- `--bucket-name`: The name of the S3 bucket containing the JSON files (**required**).
- `--query-batch-size`: The number of records to retrieve in a single database query (defaults to **1,000,000**).
- `--lambda-record-size`: The number of records to process in a single Lambda invocation (defaults to **100**).
- `--use-replica`: Flag indicating whether to use the replica database (defaults to `False`).

Here are some examples:

------------------------------------------------------------------------------------------

#### 1. Import filings/caselaw using the defaults:

- For filings: 
```bash
python manage.py make_aws_manifest_files --record-type rd --bucket-name dev-bulk-customer-data
```

- For caselaw: 
```bash
python manage.py make_aws_manifest_files --record-type o --bucket-name dev-bulk-customer-data
```

The previous commands will use the prod database to get the records. Manifest files will be limited to 1 million records each and each Lambda invocation will process a batch of 100 records at a time

#### 2. Import filings from the replica:

```bash
python manage.py make_aws_manifest_files --record-type rd --bucket-name dev-bulk-customer-data --use-replica
```

The previous commands will use the replica database to get the records. Manifest files will be limited to 1 million records each and each Lambda invocation will process a batch of 100 records at a time.

#### 3. Import filings from the replica and increase the number of records within each manifest:

```bash
python manage.py make_aws_manifest_files --record-type rd --bucket-name dev-bulk-customer-data  --query-batch-size 5000000 --use-replica
```
Manifest files will be limited to 5 million records each. However, each Lambda invocation will still process a batch of 100 records at a time.

### CODE REVIEW

File: cl/corpus_importer/management/commands/customer_import_project.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+            r.hset(
+                f"{record_type}_import_status",
+                "next_iteration_counter",
+                counter,
+            )
+            r.hset(
+                f"{record_type}_import_status",
+                "records_processed",
+                records_processed + record_count,
+            )
```
Comment (mlissner) [inline cl/corpus_importer/management/commands/customer_import_project.py:233]: You can do this in a single command:

```suggestion
            r.hset(
                f"{record_type}_import_status", 
                mapping={
                    "last_pk": last_pk,
                    "next_iteration_counter": counter,
                    "records_processed": records_processed + record_count,
                },
            )
```

File: cl/corpus_importer/management/commands/customer_import_project.py
Comment (mlissner) [inline cl/corpus_importer/management/commands/customer_import_project.py:1]: Maybe a better name would be make_aws_manifest_files.py?

**Review by mlissner — APPROVED** [review]
Two little things, but looks good to me!

File: cl/corpus_importer/management/commands/customer_import_project.py
Comment (ERosendo) [reply cl/corpus_importer/management/commands/customer_import_project.py:1]: You're right. Thanks!

File: cl/corpus_importer/management/commands/customer_import_project.py (hunk 1 above)
Comment (ERosendo) [reply cl/corpus_importer/management/commands/customer_import_project.py:233]: Thanks for the suggestion. I'll update the code

**Review by mlissner — APPROVED** [review]

**Pull request merged**